dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESReadOnlyMonitor.java

```diff
@@ -0,0 +1,192 @@
+        }
+    }
+
+    /**
+     * Start a {@link ESReadOnlyMonitor} is it is not started yet
+     * @return false if a ESReadOnlyMonitor was started before
+     */
+    public boolean start(){
+        if (started.compareAndSet(false, true)) {
+            if (ElasticsearchUtil.isClusterInReadOnlyMode()) {
+                ReindexThread.setCurrentIndexReadOnly(true);
+                sendMessage("es.cluster.read.only.message");
+                startClusterMonitor();
+            } else if (ElasticsearchUtil.isEitherLiveOrWorkingIndicesReadOnly()) {
+                ReindexThread.setCurrentIndexReadOnly(true);
+                sendMessage("es.index.read.only.message");
+                startIndexMonitor();
+            } else {
+                started.set(false);
+            }
+
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    private void sendMessage(final String messageKey) {
+        try {
+            final Role adminRole = roleAPI.loadCMSAdminRole();
+            final List<String> usersId = roleAPI.findUsersForRole(adminRole)
+                    .stream()
+                    .map(user -> user.getUserId())
+                    .collect(Collectors.toList());
+
+            final String message = LanguageUtil.get(messageKey);
+
+            final SystemMessageBuilder messageBuilder = new SystemMessageBuilder()
+                    .setMessage(message)
+                    .setSeverity(MessageSeverity.ERROR)
+                    .setType(MessageType.SIMPLE_MESSAGE)
+                    .setLife(TimeUnit.SECONDS.toMillis(5));
+
+            Logger.error(ESReadOnlyMonitor.class, message);
+            systemMessageEventUtil.pushMessage(messageBuilder.create(), usersId);
+        } catch (final LanguageException | DotDataException | DotSecurityException e) {
+            Logger.warn(ESReadOnlyMonitor.class, () -> e.getMessage());
+        }
+    }
+
+    private void putCurrentIndicesToWriteMode() {
+        try {
+            Logger.debug(this.getClass(), () -> "Trying to set the current indices to Write mode");
+            ElasticsearchUtil.setLiveAndWorkingIndicesToWriteMode();
+            sendMessage("es.index.write.allow.message");
+            ReindexThread.setCurrentIndexReadOnly(false);
+
+            this.stop();
+        } catch (final ElasticsearchResponseException e) {
+            Logger.info(ESReadOnlyMonitor.class, ()  -> e.getMessage());
+        }
+    }
+
+    private void putClusterToWriteMode() {
+        Logger.debug(this.getClass(), () -> "Trying to set the current indices to Write mode");
+        ElasticsearchUtil.setClusterToWriteMode();
+        sendMessage("es.cluster.write.allow.message");
+        ReindexThread.setCurrentIndexReadOnly(false);
+
+        this.stop();
+    }
+
+    private void startIndexMonitor() {
+         schedule(new IndexMonitorTimerTask(this));
+    }
+
+    private synchronized void schedule(final TimerTask timerTask) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use block level rather than method level synchronization](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5642172)